mmv1/products/networksecurity/UrlLists.yaml

```diff
@@ -0,0 +1,103 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Resource
+name: "UrlLists"
+base_url: "projects/{{project}}/locations/{{location}}/urlLists"
+create_url: "projects/{{project}}/locations/{{location}}/urlLists?urlListId={{name}}"
+self_link: "projects/{{project}}/locations/{{location}}/urlLists/{{name}}"
+min_version: beta
+update_verb: :PATCH
+update_mask: true
+description: |
+  UrlList proto helps users to set reusable, independently manageable lists of hosts, host patterns, URLs, URL patterns.
+references:
+  !ruby/object:Api::Resource::ReferenceLinks # TODO(diogoesteves): change the url to beta once it is available.
+  guides:
+    "Official Documentation": "https://cloud.google.com/secure-web-proxy/docs/use-url-list"
+  api: "https://cloud.google.com/secure-web-proxy/docs/reference/network-security/rest/v1alpha1/projects.locations.urlLists"
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 30
+      update_minutes: 30
+      delete_minutes: 30
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+autogen_async: true
+import_format: ["projects/{{project}}/locations/{{location}}/urlLists/{{name}}"]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: "network_security_url_lists_basic"
+    primary_resource_id: "default"
+    vars:
+      resource_name: "my-url-lists"
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: "network_security_url_lists_advanced"
+    primary_resource_id: "default"
+    vars:
+      resource_name: "my-url-lists"
+parameters:
```

Comment:
If name or location changes, it should create a new resource, right? If that is true, `immutable: true` should be added for those two properties.

Reply:
Oh great, you are correct! Added the immutable for name and location.